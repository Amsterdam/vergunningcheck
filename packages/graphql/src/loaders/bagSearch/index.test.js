require("@spotify/polly-jest-presets");
const { loaderFunctionsTest } = require("../../testUtils");
const mock = require("./__mocks__/basisweg30.json");

const bagSearch = require(".");

const key1 = "basisweg 30";
const key2 = "1055xd 19c";

describe("BagSearch loader", () => {
  describe("Basic functions", loaderFunctionsTest(bagSearch, key1, key2));

  test("reducer", async () => {
    expect(bagSearch.reducer(mock)).toMatchSnapshot();
  });

  test("findAddress", async () => {
    const x = await bagSearch.load(["basisweg 30"]);
    expect(await x[0]).toMatchSnapshot();
  });
});
