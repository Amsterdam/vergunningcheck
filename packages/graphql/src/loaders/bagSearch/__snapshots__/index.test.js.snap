// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BagSearch loader Basic functions module result 1`] = `
Array [
  Array [
    Object {
      "_adressableObjectId": "0363010000578090",
      "houseNumber": 30,
      "houseNumberFull": "30",
      "houseNumberLetter": null,
      "houseNumberSuffix": null,
      "postalCode": "1043AP",
      "residence": "Amsterdam",
      "streetName": "Basisweg",
      "type": "BUILDING",
    },
  ],
]
`;

exports[`BagSearch loader Basic functions module result 2`] = `
Array [
  Array [
    Object {
      "_adressableObjectId": "0363010012062064",
      "houseNumber": 19,
      "houseNumberFull": "19 C",
      "houseNumberLetter": "C",
      "houseNumberSuffix": null,
      "postalCode": "1055XD",
      "residence": "Amsterdam",
      "streetName": "Louise de Colignystraat",
      "type": "BUILDING",
    },
  ],
]
`;

exports[`BagSearch loader findAddress 1`] = `
Array [
  Object {
    "_adressableObjectId": "0363010000578090",
    "houseNumber": 30,
    "houseNumberFull": "30",
    "houseNumberLetter": null,
    "houseNumberSuffix": null,
    "postalCode": "1043AP",
    "residence": "Amsterdam",
    "streetName": "Basisweg",
    "type": "BUILDING",
  },
]
`;

exports[`BagSearch loader findAddress 2`] = `
Array [
  Array [
    Object {
      "_adressableObjectId": "0363010012062064",
      "houseNumber": 19,
      "houseNumberFull": "19 C",
      "houseNumberLetter": "C",
      "houseNumberSuffix": null,
      "postalCode": "1055XD",
      "residence": "Amsterdam",
      "streetName": "Louise de Colignystraat",
      "type": "BUILDING",
    },
  ],
]
`;

exports[`BagSearch loader findAddress 3`] = `
Array [
  Object {
    "_adressableObjectId": "0363010000578090",
    "houseNumber": 30,
    "houseNumberFull": "30",
    "houseNumberLetter": null,
    "houseNumberSuffix": null,
    "postalCode": "1043AP",
    "residence": "Amsterdam",
    "streetName": "Basisweg",
    "type": "BUILDING",
  },
]
`;

exports[`BagSearch loader reducer 1`] = `
Object {
  "_adressableObjectId": "0363010000578090",
  "houseNumber": 30,
  "houseNumberFull": "30",
  "houseNumberLetter": null,
  "houseNumberSuffix": null,
  "postalCode": "1043AP",
  "residence": "Amsterdam",
  "streetName": "Basisweg",
  "type": "BUILDING",
}
`;
